src/CodeGen_LLVM.cpp

```diff
@@ -1739,13 +1875,7 @@ Value *CodeGen_LLVM::codegen_buffer_pointer(Value *base_address, Halide::Type ty
         index = promote_64(index);
     }
 
-    // Handles are always indexed as 64-bit.
-    if (type.is_handle()) {
-        return codegen_buffer_pointer(base_address, UInt(64, type.lanes()), index);
-    } else {
```

Comment:
Not sure I understand why this case won't arise anymore.

Reply:
It's now handled in the codegen_buffer_pointer variant that this calls. upgrade_type_for_storage does the promotion of handles to uint64s.